File: advent01a.dart
// Advent of Code 2017
// Day 1 - Part I
// Dart


import 'dart:async';
import 'dart:io';

void main() {
  new File("advent01.in").readAsString().then((String allFileStr) {
      int sumTotal = 0;
      for (int i = 0; i < allFileStr.length; i++) {
          String thisChar = allFileStr[i];
          String nextChar = allFileStr[(i + 1) % allFileStr.length];
          if (thisChar == nextChar) sumTotal += int.parse(thisChar);
      }

      print("Total: ${sumTotal}");
  });
}

File: advent01a.py
# Advent of Code 2017
# Day 1 - Part I
# Python


f = open("advent01.in", "r")
allFileStr = f.read()

sumTotal = 0  # To keep track of total.
iter = 0  # To keep track of iterator.
for thisChar in allFileStr:  # Probably could have used a standard for loop, since I'm using an iterator, but whatever.
    nextChar = allFileStr[(iter + 1) % len(allFileStr)]  # Using the modulo to wrap the value back to the beginning.
    if thisChar == nextChar:
        sumTotal += int(thisChar)
    iter += 1


print("TOTAL: " + sumTotal)
